src/queries/searchResultQuery.ts

```diff
@@ -0,0 +1,66 @@
+import BaseQuery, { QueryParams } from './BaseQuery';
+
+interface SearchResultQueryParams extends QueryParams {
+  keyword?: string;
+  urlEquals?: string;
+  userId?: number;
+}
+
+class SearchResultQuery extends BaseQuery {
+  keyword?: string;
+  urlEquals?: string;
+  userId?: number;
+  whereQuery: { [index: string]: any };
+
+  constructor(searchResultQueryParams: SearchResultQueryParams) {
+    super(searchResultQueryParams);
+
+    this.keyword = searchResultQueryParams.keyword;
+    this.urlEquals = searchResultQueryParams.urlEquals;
+    this.userId = searchResultQueryParams.userId;
+    this.whereQuery = {};
+  }
+
+  async call() {
+    this.filterByUserId();
+    this.filterByKeyword();
+    this.filterByUrl();
+    this.paginate();
+
+    return this.databaseInstance.findMany({
+      where: { ...this.whereQuery },
+      ...this.paginationQueryOptions,
```

Comment:
How about:

```suggestion
      ...this.paginationQuery,
```

`whereQuery` && `paginationQuery` may sounds better together? 👀